### ISSUE TITLE
Select Text button in smartbar

### ISSUE THREAD

**Issue by goyalyashpal** [florisboard/florisboard#352]
Split from https://github.com/florisboard/florisboard/issues/341 (1)

#### Short description
Ttoggle select mode" can be added to smartbar in my opinion. With toggle select mode, i meant this button in text editor mode. There is just enough space for one more icon to come there.

![zjw3NnJ7ec](https://user-images.githubusercontent.com/19423063/107872959-6ca7f200-6ed4-11eb-89cf-5643db8d680d.png)

---
**Rough mockup:** \
the icon i chose is basically a combination of area icon ([see example here🡭](https://www.iconfinder.com/icons/6602651/area_design_graphic_select_selection_tool_icon)) and text cursor. The circle underneath was used to ensure equal spacing.
![photo_2021-02-12_23-30-39 (2)](https://user-images.githubusercontent.com/19423063/107873454-b2ff5000-6ed8-11eb-96b0-9316fde7f4f2.jpg)

---
**Use case:** 
i use an app called hovetext a lot. And it is VERY cumbersome to select text there as it doesnt show the android hovering text select options, rather it displays its own static toolbar which occupies the limited screen space - which becomes a recurring papercut. So, having quick access to enable select mode will be veryy very handy to me.

![image](https://user-images.githubusercontent.com/19423063/107874887-d3340c80-6ee2-11eb-96f3-1beabd45ec2f.png)

**Comment by Glitchy-Tozier** [comment]
Just a little heads-up: You might also be able to use a long-press or gesture to switch to the selection keyboard.

**Comment by yashpalgoyal1304** [comment]
> I really like the icon you created.

thanks :)

**Comment by yashpalgoyal1304** [comment]
> Just a little heads-up: You might also be able to use a long-press or gesture to switch to the selection keyboard.

i couldnt find that. Where?

**Comment by Glitchy-Tozier** [comment]
> > Just a little heads-up: You might also be able to use a long-press or gesture to switch to the selection keyboard.
> 
> i couldnt find that. Where?

I think i made that up... I vaguely remembered a discussion on the topic, but apparently it never was inplemented.

**Comment by patrickgold** [comment]
> I think i made that up... I vaguely remembered a discussion on the topic, but apparently it never was inplemented.

We talked about different toolbars in the Smartbar - and that you can switch between them via gestures (which is still on the todo list). And we also talked about the fact that the current selection keyboard is very inaccessible. So that's the reason I think you vaguely remember.

**Comment by MisterY** [comment]
> Edit2: While I still think that the button shouldn't be in the copy&paste-bar and should instead be made accessible through different means, I really like the icon you created.

Why is there a Select All button, then? That can also "be made accessible through different means". Or Cut, Copy, and Paste, as well. Right?

**Comment by Glitchy-Tozier** [comment]
1. The selection-board is a 
 view/keyboard. That's different from a single functionality. And yes, I don't think adding a gesture to switch to certain views/keyboards (like the selection-board) is a bad idea.
2. The problem I see is this: All the current features of the copy&paste-bar synergize nicely with each other.
Compare that to the selection-board, which does the exact same things all over again, making placing it in the copy&paste-strip almost completely redundant.

**Comment by MisterY** [comment]
Thanks for the tip. I didn't realise that there is the whole separate selection screen. The icon looks like the keyboard floating or movement mode, so I didn't try it.
true, that makes the whole selection bar a bit redundant. But the icon to access it should then be in the first bar, not only accessible after additional action. 

Nonetheless, the point I'm making has nothing to do with the selection screen but the bar. If there is Select All, the cursor movement keys, plus all the usual clipboard operations, then the Select Text belongs there, too.
With the selection screen in place, though, all of these are redundant and should leave space for other, more important options. In thiscase , I'd probably prefer to have the second smart bar buttons as the default (smileys, options, selection button, etc.)

**Comment by yashpalgoyal1304** [comment]
> and should leave space for other, more important options. In this case , I'd probably prefer to have the second smart bar buttons as the default (smileys, options, selection button, etc.)

hi @MisterY would u like to open a new issue for this suggestion? And put it in continuation of #341 > 1B

> So, if that is the case, smart bar may be made customisable i.e. which tools/icons to show there be user choosable.
_It is not split per se, as it seems out of scope currently. And i dont have interest in it either - if it's not required for the purposes mentioned in the basics section above.
But feel free to open this as a separate one if u want this, but mention this issue's link there/or mention the new issue's link here_

**Comment by yashpalgoyal1304** [comment]
with put in continuation, i meant that mention in the starting (like it's done in starting of this issue) that:
" **Split from #341 (1B)** "

**Comment by MisterY** [comment]
I added a few new proposals, thanks!

**Comment by patrickgold** [comment]
As of v0.4.6 there's full support of shift+arrow keys for selecting text, as such an additional text selection button is not needed anymore in our opinion. Closing thus.

**Comment by goyalyashpal** [comment]
hey @patrickgold !  did you mean:
* #1843 ?
* florisboard@0.3.16 https://github.com/florisboard/florisboard/releases/tag/v0.3.16

**Comment by patrickgold** [comment]
No, i meant shift + smartbar artow keys (they are by default in the extended smartbar but you can reorder it)
